core/shared/utils/bh_log.c

```diff
@@ -79,3 +79,29 @@ bh_print_time(const char *prompt)
 
     last_time_ms = curr_time_ms;
 }
+
+void
+bh_print_proc_mem(const char *prompt)
+{
+    char buf[1024] = { 0 };
+
+    if (log_verbose_level < BH_LOG_LEVEL_DEBUG)
+        return;
+
+    if (os_dumps_proc_mem_info(buf, sizeof(buf)) != 0)
+        return;
+
+    os_printf("%s\n", prompt);
+    os_printf("===== /proc/self/status =====\n");
```

Comment:
This won't always be a buffer from `/proc/self/status` I guess? (e.g. not in windows) I suggested another approach for this, see my comment above.